OrbitCore/Callstack.h

```diff
@@ -7,20 +7,24 @@
 
 #include <xxhash.h>
 
-#include <string>
 #include <vector>
 
 #include "CallstackTypes.h"
 
 struct CallStack {
   CallStack() = default;
-  inline CallstackID Hash() {
-    if (m_Hash != 0) return m_Hash;
-    m_Hash = XXH64(m_Data.data(), m_Data.size() * sizeof(uint64_t), 0xca1157ac);
-    return m_Hash;
-  }
+  explicit CallStack(std::vector<uint64_t>&& addresses) {
+    data = std::move(addresses);
+    hash = XXH64(data.data(), data.size() * sizeof(uint64_t), 0xca1157ac);
```

Comment:
Would it make sense to initialize it lazily the first time `GetHash` is called?

Reply:
this makes GetHash non-const and also doesn't save us much or does it?

Reply:
`hash_` would be `mutable` as it's a cached value. Or it's fine as is.

Reply:
I don't think that we use lazy initialization very often, and I would prefer to initialize hash in the constructor then